spec/unit/dotnet_agent_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+require 'fauxhai'
+
+describe 'appdynamics::dotnet_agent' do
+  context 'win2008r2' do
+    before do
+      Fauxhai.mock('platform' => 'windows', 'version' => '2008R2')
+    end
+    let(:chef_run) do
+      ChefSpec::SoloRunner.new do |node|
```

Comment:
Does ChefSpec not run Fauxhai for you if you pass `platform`, etc. to `SoloRunner.new`?

```
ChefSpec::SoloRunner.new(platform: 'windows', version: '2008R2') do |node|
  # ...
end
```